lib/HeroBonus.cpp

```diff
@@ -493,45 +495,33 @@ int BonusList::totalValue() const
 			break;
 		}
 	}
-	for(auto src : sources)
+	for(const auto & src : sources)
 	{
-		any.base += percent(src.base ,src.percentToSource);
+		any.base += percent(src.base, src.percentToSource);
 		any.percentToBase += percent(src.percentToBase, src.percentToSource);
 		any.percentToAll += percent(src.percentToAll, src.percentToSource);
 		any.additive += percent(src.additive, src.percentToSource);
 		if(hasIndepMin)
 			vstd::amin(any.indepMin, percent(src.indepMin, src.percentToSource));
 		if(hasIndepMax)
-			vstd::amax(any.indepMax, percent(src.indepMin, src.percentToSource));
+			vstd::amax(any.indepMax, percent(src.indepMax, src.percentToSource));
 	}
 	any.base = percent(any.base, any.percentToBase);
 	any.base += any.additive;
 	auto valFirst = percent(any.base ,any.percentToAll);
 
-	if(hasIndepMin && hasIndepMax)
-		assert(any.indepMin < any.indepMax);
+	if(hasIndepMin && hasIndepMax && any.indepMin < any.indepMax)
+		any.indepMax = std::min(any.indepMax, any.indepMin);

```

Comment:
Redundant check?
Since conditional includes `any.indepMin < any.indepMax` then std::min is redundant - you can simply use `any.indepMax = any.indepMin`

Also not sure what this supposed to do... 
If this code should handle case where min > max then it won't work due to condition

Reply:
@IvanSavenko yes, this code will handle such changes, because now INDEPENDENT_MIN means AT_MOST, and INDEPENDENT_MAX means AT_LEAST (and it was be from introduction of this properties and used in all mods).

Reply:
removed std::max

Reply:
Example: recanter's cloack from (master branch)[https://github.com/vcmi/vcmi/blob/master/config/artifacts.json#L1192]
```json
	"recantersCloak":
	{
		"index" : 83,
		"type" : ["HERO"],
		"bonuses": [
			{
				"type" : "BLOCK_MAGIC_ABOVE",
				"val" : 2,
				"valueType" : "INDEPENDENT_MIN",
				"propagator": "BATTLE_WIDE"
			}
		]
	},
```

But we cleanly have blocked spells from level 3, 4, 5, and not 1 and 2.

Reply:
Angel wings and Fly spell:

```json
	"angelWings":
	{
		"bonuses" : [
			{
				"type" : "FLYING_MOVEMENT",
				"val" : 0,
				"valueType" : "INDEPENDENT_MIN"
			}
		],
		"index" : 72,
		"type" : ["HERO"]
	},
```
and
```json
				"effects" : {
					"fly" : {
						"type" : "FLYING_MOVEMENT",
						"duration" : "ONE_DAY",
						"val" : 40,
						"valueType" : "INDEPENDENT_MIN"
					}
				}
```
So, if you cast basic Fly, then angel wings will override it, because FLYING_MOVEMENT can be AT MOST 0, not higher.